### PR TITLE
Search SDK: Fixing unit tests that were broken by VS2017 migration

### PR DESCRIPTION
**Pull request by bitsforthought**
As part of moving to VS2017, the Search SDK is using a newer version of the ClientRuntime. Two of the unit tests were pinning constructor validation behavior of ServiceClient, which seems to have changed.

Also, the move to VS2017 required using a newer version of the Resource Manager SDK, which required re-recording all mocks.

Finally, for the data plane SDK, we upgraded to Microsoft.Spatial 7.2.0, which is compatible with VS2017 and .NET Core.

FYI @shahabhijeet

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [X] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Spatial" Version="6.15.0" />
+    <PackageReference Include="Microsoft.Spatial" Version="7.2.0" />
```
Comment (shahabhijeet) [inline src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Microsoft.Azure.Search.csproj]: @brjohnstmsft please remove all commented code from .csproj files. They were left for reference purpose during migration. If things are building correctly, they should be removed.
Also remove any traces of project.json, global.json and *.xproj, these files are no longer required.

File: src/SDKs/Search/DataPlane/Search.Tests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -3,8 +3,8 @@
 // license information.
 
 using System.Reflection;
-using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using Xunit;
```
Comment (shahabhijeet) [inline src/SDKs/Search/DataPlane/Search.Tests/Properties/AssemblyInfo.cs]: @brjohnstmsft any reason this is required in AssemblyInfo.cs? If not please remove.

File: src/SDKs/Search/Management/Search.Management.Tests/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 using System.Reflection;
 using System.Runtime.InteropServices;
+using Xunit;
```
Comment (shahabhijeet) [inline src/SDKs/Search/Management/Search.Management.Tests/Properties/AssemblyInfo.cs]: @brjohnstmsft if not required, please remove

File: src/SDKs/Search/Management/Search.Management.Tests/Search.Management.Tests.csproj [hunk 4]
```diff
@@ -12,7 +12,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.1.1-preview" />
+    <!-- <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.1.1-preview" /> -->
```
Comment (shahabhijeet) [inline src/SDKs/Search/Management/Search.Management.Tests/Search.Management.Tests.csproj]: @brjohnstmsft if you are re-recording why are you using an older version of resource manager nuget package.
You should be using the latest. I believe we did talk about it.
The implicit resource manager package is for legacy tests that are still using it. The sooner we move away from the legacy package the better it is.

File: src/SDKs/dirs.proj [hunk 5]
```diff
@@ -36,6 +36,8 @@
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Relay\Management.Relay\Microsoft.Azure.Management.Relay.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Resource\Management.ResourceManager\Microsoft.Azure.Management.ResourceManager.csproj" />
 		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Scheduler\Management.Scheduler\Microsoft.Azure.Management.Scheduler.csproj" />
+		<ProjectToBuild Include="$(MSBuildThisFileDirectory)Search\Management\Management.Search\Management.Search.csproj" />
```
Comment (shahabhijeet) [inline src/SDKs/dirs.proj]: @brjohnstmsft these files are no longer needed.
These are kept as a backup incase we need to switch back to declarative model for building projects.
But as of today, no changes are required for this file, but will not break anything if you make any updates.
Do not revert, just as an FYI

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Search/Management/Search.Management.Tests/Search.Management.Tests.csproj (hunk 4 above)
Comment (brjohnstmsft) [inline src/SDKs/Search/Management/Search.Management.Tests/Search.Management.Tests.csproj]: I thought the implicit one was the new one. Now I have to re-record again. :-(

File: src/SDKs/dirs.proj (hunk 5 above)
Comment (brjohnstmsft) [inline src/SDKs/dirs.proj]: Ok

**Comment by brjohnstmsft** [comment]
I'm going to close this and open a new one after rebasing my changes.

**Pull request closed without merging**